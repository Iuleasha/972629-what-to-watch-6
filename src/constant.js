export const HeaderMode = {
  MOVIE_CARD: `movie-card__head`,
  USER_PAGE: `user-page__head`,
};

export const PosterSize = {
  WIDTH: 218,
  HEIGHT: 327,
};

export const MovieCardButtonSize = {
  WIDTH: 19,
  HEIGHT: 20,
};

export const MOVIE_CARD_NAV_ITEMS = [`Overview`, `Details`, `Reviews`];

export const DEFAULT_GENRE = `All genres`;

export const AppRoute = {
  LOGIN: `/login`,
  LOSE: `/lose`,
  MY_LIST: `/mylist`,
  ROOT: `/`,
  FILMS: `/films`,
  FILM: `/films/:id`,
  PLAYER: `/player/:id`,
  REVIEW: `/films/:id/review`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const APIRoute = {
  FILMS: `/films`,
  FILM: `/film`,
  PROMO_FILM: `/films/promo`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`
};
