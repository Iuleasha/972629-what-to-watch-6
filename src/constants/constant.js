export const HeaderMode = {
  MOVIE_CARD: `movie-card__head`,
  USER_PAGE: `user-page__head`,
};

export const PosterSize = {
  WIDTH: 218,
  HEIGHT: 327,
};

export const MovieCardButtonSize = {
  WIDTH: 19,
  HEIGHT: 20,
};

export const MOVIE_CARD_NAV_ITEMS = [`Overview`, `Details`, `Reviews`];

export const DEFAULT_GENRE = `All genres`;

export const AppRoute = {
  LOGIN: `/login`,
  LOSE: `/lose`,
  MY_LIST: `/mylist`,
  ROOT: `/`,
  FILMS: `/films`,
  FILM: `/films/:id`,
  PLAYER: `/player/:id`,
  REVIEW: `/films/:id/review`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const APIRoute = {
  FILMS: `/films`,
  FILM: `/film`,
  PROMO_FILM: `/films/promo`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`
};

export const ButtonIcon = {
  PLAY: `#play-s`,
  PAUSE: `#pause`,
  FULL_SCREEN: `#full-screen`,
  INLIST: `#in-list`,
  ADD: `#add`,
};

export const NameSpace = {
  DATA: `DATA`,
  COMMENTS: `COMMENTS`,
  USER: `USER`,
};

export const ActionType = {
  LOAD_SUCCESS: `load/loadSuccess`,
  LOAD_PROMO_FILM: `load/loadPromoFilm`,
  SET_GENRE: `filter/genres`,
  LOAD_FAVORITE: `load/favorite`,
  SWITCH_FAVORITE_STATUS: `load/switchFavoriteStatus`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  REDIRECT_TO_ROUTE: `site/redirectToRoute`,
  SET_USER: `user/setUser`,
  ADD_REVIEW: `film/addReview`,
};

export const FILMS_PAGE_SIZE = 8;

export const AVATAR_SIZE = 63;

export const CommentLength = {
  MIN: 50,
  MAX: 400
};

export const STARS_LENGTH = 10;

export const RATING_DEFAULT_VALUE = 3;

export const RATING_STARS = [...Array(STARS_LENGTH).keys()].map((i) => i + 1);

export const Rating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};
