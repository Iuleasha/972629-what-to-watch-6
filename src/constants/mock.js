export const FILM_MOCK = {
  'id': 1,
  'name': `The Grand Budapest Hotel`,
  'poster_image': `img/the-grand-budapest-hotel-poster.jpg`,
  'preview_image': `img/the-grand-budapest-hotel.jpg`,
  'background_image': `img/the-grand-budapest-hotel-bg.jpg`,
  'background_color': `#ffffff`,
  'video_link': `https://some-link`,
  'preview_video_link': `https://some-link`,
  'description': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  'rating': 8.9,
  'scores_count': 240,
  'director': `Wes Andreson`,
  'starring': [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  'run_time': 99,
  'genre': `Comedy`,
  'released': 2014,
  'is_favorite': false,
};

export const COMMENT_MOCK = {
  'id': 1,
  'user': {
    'id': 4,
    'name': `Kate Muir`,
  },
  'rating': 8.9,
  'comment': `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  'date': `2019-05-08T14:13:56.569Z`,
};

export const AUTH_INFO_MOCK = {
  'id': 1,
  'email': `Oliver.conner@gmail.com`,
  'name': `Oliver.conner`,
  'avatar_url': `img/1.png`,
};
