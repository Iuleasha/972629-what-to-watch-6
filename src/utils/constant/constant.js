export const HeaderMode = {
  MOVIE_CARD: `movie-card__head`,
  USER_PAGE: `user-page__head`,
};

export const PosterSize = {
  WIDTH: 218,
  HEIGHT: 327,
};

export const MovieCardButtonSize = {
  WIDTH: 19,
  HEIGHT: 20,
};

export const DEFAULT_GENRE = `All genres`;
