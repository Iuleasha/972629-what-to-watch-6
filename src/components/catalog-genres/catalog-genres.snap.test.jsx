import {render} from '@testing-library/react';
import React from 'react';
import {DEFAULT_GENRE} from '../../constants/constant';
import CatalogGenres from './catalog-genres';

test(`Should CatalogGenres render correctly`, () => {
  const {container} = render(<CatalogGenres activeFilter={DEFAULT_GENRE} genres={[DEFAULT_GENRE]} onSelectGenre={() => {
  }}/>);
  expect(container).toMatchSnapshot();
});
