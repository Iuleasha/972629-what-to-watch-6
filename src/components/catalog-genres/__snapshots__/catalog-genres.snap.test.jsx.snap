// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CatalogGenres render correctly 1`] = `
<div>
  <ul
    class="catalog__genres-list"
  >
    <li
      class="catalog__genres-item catalog__genres-item--active"
    >
      <a
        class="catalog__genres-link"
      >
        All genres
      </a>
    </li>
  </ul>
</div>
`;
